topdown/lookup_test.go

```diff
@@ -0,0 +1,67 @@
+package topdown
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestLookup(t *testing.T) {
+	cases := []struct {
+		note     string
+		object   string
+		key      interface{}
+		fallback interface{}
+		expected interface{}
+	}{
+		{
+			note:     "basic case . found",
+			object:   `{"a": "b"}`,
+			key:      `"a"`,
+			fallback: `"c"`,
+			expected: `"b"`,
+		},
+		{
+			note:     "basic case . not found",
+			object:   `{"a": "b"}`,
+			key:      `"c"`,
+			fallback: `"c"`,
+			expected: `"c"`,
+		},
+		{
+
+			note:     "integer key . found",
+			object:   "{1: 2}",
```

Comment:
Aren't non-string keys a bit of an edge case? From what I can tell, non-string object keys work _only until you print them_. (Like `y := { x: true | x := [1, 2, 3]}` or your `y := { 1: 2}`)

Anyhow, I *think* this is in line with the other builtins... 🤔

Reply:
That's correct. Internally it works fine but when values have to be converted into JSON it will error. As is, the built-in is doing the right thing.

Reply:
Yeah I originally implemented this only supporting the string key but then figured out that if I'm actually more liberal, there is actually less code to be written.